packages/gatsby-cli/src/reporter/loggers/json/index.ts

```diff
@@ -0,0 +1,28 @@
+import { onLogAction } from "../../redux/index"
+import { ActionsUnion, ISetStatus } from "../../redux/types"
+import stripAnsi from "strip-ansi"
+import _ from "lodash"
+
+const isStringPayload = (action: ActionsUnion): action is ISetStatus =>
+  typeof action.payload === `string`
+
+const sanitizeAction = (action: ActionsUnion): ActionsUnion => {
+  if (isStringPayload(action)) {
+    return action
+  }
+
+  if (`text` in action.payload && action.payload.text) {
+    action.payload.text = stripAnsi(action.payload.text)
+  }
+  if (`statusText` in action.payload && action.payload.statusText) {
+    action.payload.statusText = stripAnsi(action.payload.statusText)
+  }
+
+  return action
```

Comment:
I think there is a functional change going on here. In the previous version, we were creating a new object and spreading values into it. This had the benefit that we weren't mutating the underlying object.

I think there is a risk here that this function in it's new state would actually be modifying values that are in redux state (thanks JS object referencing.)

Let's change this to create a new object instead of modifying the one passed in.

Reply:
Now I use guideline from IPC to copy new object before `stripAnsi`.